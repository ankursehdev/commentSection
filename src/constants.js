export const commentDetails = {
  authorName: "Brad",
  author: "Author",
  commentText:
    "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Id veniam quia distinctio temporibus delectus laborum, error qui iusto recusandae nesciunt voluptatem cupiditate.",
  time: "2 Minutes ago",
  numberOfReplies: 21,
  upVotes: 123,
  downVotes: 0,
};
