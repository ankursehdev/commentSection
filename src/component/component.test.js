import React from "react";
import { render, cleanup } from "@testing-library/react";
import { commentDetails } from "../constants";

import Comment from "./index";
afterEach(cleanup);

test("matches snapshot", () => {
  const { asFragment } = render(<Comment commentDetails={commentDetails} />);
  expect(asFragment()).toMatchSnapshot();
});

test("Checking the comment text", () => {
  const { getByTestId } = render(<Comment commentDetails={commentDetails} />);

  const commentText = getByTestId("comment-text");
  expect(commentText).toBeInTheDocument();
});
