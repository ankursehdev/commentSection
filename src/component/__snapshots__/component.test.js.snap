// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="comment-wrapper"
  >
    <section
      class="comment-section"
    >
      <figure
        class="avatar"
      >
        <img
          alt="Profile"
          class="avatar-image"
          src="https://cdn-images-1.medium.com/fit/c/72/72/0*B0gTOtTQdksop1vG."
        />
      </figure>
      <section
        class="content-section"
      >
        <header
          class="content-section-header"
          data-header="header"
        >
          <h4
            class="author-name"
          >
            Brad
          </h4>
          <span
            class="commentor"
          >
            Author
          </span>
          <time
            class="time"
          >
            <small>
              ‚óè
            </small>
             2 Minutes ago
          </time>
        </header>
        <div
          class="comment-body"
        >
          <p
            data-testid="comment-text"
          >
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Id veniam quia distinctio temporibus delectus laborum, error qui iusto recusandae nesciunt voluptatem cupiditate.
          </p>
        </div>
        <div
          class="actions"
        >
          <button>
            Reply
          </button>
          <button>
            <span
              class="reply-count"
            >
              21
            </span>
             Replies
          </button>
          <button
            class="vote-up"
          >
            123
          </button>
          <button
            class="vote-down"
          >
            0
          </button>
        </div>
      </section>
    </section>
  </div>
</DocumentFragment>
`;
